task_processing/interfaces/task_executor.py

```diff
@@ -24,14 +21,14 @@ class TaskExecutor(object):
     """
 
     @abc.abstractmethod
-    def run(self, task_config):
-        """Run the supplied task
+    def run(self, task_config, task_id):
```

Comment:
I think this is where the disagreement is. I am advocating for a solution that only involves changing the retrying executor. 
Something like this: https://gist.github.com/sagar8192/bb98d0b479b3cf1afcf09785529471fd

This way the translation logic lives only in one layer. Other layers do not need to know anything about details of the retrying executor(do not need to join task_id etc). We can add logging to make sure that task_id mapping info is captured for debugging. IMO, this change is less intrusive and also does not introduce complexity in other layers.